rclpy/rclpy/executors.py

```diff
@@ -511,50 +510,32 @@ def _wait_for_ready_callbacks(
             with ExitStack() as context_stack:
                 sub_handles = []
                 for sub in subscriptions:
-                    try:
-                        context_stack.enter_context(sub.handle)
-                        sub_handles.append(sub.handle)
-                    except InvalidHandle:
-                        entity_count.num_subscriptions -= 1
+                    context_stack.enter_context(sub.handle)
+                    sub_handles.append(sub.handle)
 
                 client_handles = []
                 for cli in clients:
-                    try:
-                        context_stack.enter_context(cli.handle)
-                        client_handles.append(cli.handle)
-                    except InvalidHandle:
-                        entity_count.num_clients -= 1
+                    context_stack.enter_context(cli.handle)
+                    client_handles.append(cli.handle)
 
                 service_handles = []
                 for srv in services:
-                    try:
-                        context_stack.enter_context(srv.handle)
-                        service_handles.append(srv.handle)
-                    except InvalidHandle:
-                        entity_count.num_services -= 1
+                    context_stack.enter_context(srv.handle)
+                    service_handles.append(srv.handle)
 
                 timer_handles = []
                 for tmr in timers:
-                    try:
-                        context_stack.enter_context(tmr.handle)
-                        timer_handles.append(tmr.handle)
-                    except InvalidHandle:
```

Comment:
I think these `try/except` blocks need to be kept because it's still possible for this exception to be raised.

Reply:
Same comment pretty much everywhere these blocks are removed in this PR